src/routes/category/[id]/+page.svelte

```diff
@@ -27,33 +29,36 @@
     $: currentPage = currentCategoryOffers.currentPage
 
     async function loadCategoryProducts(categoryId: string) {
-        await executeAction(isLoading, fetchCategoryProducts, categoryId)
+        await fetchCategoryProductsHandler(categoryId)
         loadedCategoryId = categoryId
     }
 
-    async function fetchCategoryProducts(categoryId: string, page?: number): Promise<void> {
-        const response = await fetch(`${API_BASE_ENDPOINT}/offer?categoryId=${categoryId}&${page ? `page=${page}&` : ''}limit=${pageLimit}`, {
-            method: 'GET',
-            headers: {...BASE_HEADERS}
-        })
+    async function fetchCategoryProductsHandler(categoryId: string, page?: number): Promise<void> {
```

Comment:
The name `fetchCategoryProductsHandler` could be confusing because of the handler. How about `handleCategoryProductsFetch`